machiavelli-cards/src/main/java/pl/jellytech/machiavelli/cards/controllers/CardController.java

```diff
@@ -0,0 +1,94 @@
+package pl.jellytech.machiavelli.cards.controllers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import pl.jellytech.machiavelli.cards.dtos.CardResponse;
+import pl.jellytech.machiavelli.cards.entities.Card;
+import pl.jellytech.machiavelli.cards.entities.CardType;
+import pl.jellytech.machiavelli.cards.services.ICardService;
+import pl.jellytech.machiavelli.cards.utils.ControllerUtils;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController()
+@RequestMapping( "api/card")
+@Slf4j
+public class CardController {
+    private final ICardService cardService;
+    @Autowired
+    public CardController(ICardService cardService){
+        this.cardService = cardService;
+    }
+    @PostMapping()
+    public ResponseEntity<CardResponse> CreateOrUpdate(@RequestParam("cardType") CardType cardType,
+                                               @RequestParam("name") String name,
+                                               @RequestParam("description") String description,
+                                               @RequestParam(value = "cardId",required = false) Optional<Long> cardId,
+                                               @RequestPart("image") MultipartFile image)
+            throws IOException {
```

Comment:
@klisowska-wlog Why i should avoid that?. If i will wrapped in custom error response (below) will be ok?

`        catch (IOException ex){
            return ControllerUtils.ErrorResponse(ex.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
        }`

Reply:
For integration purposes mostly. In this internal app it's not that important since we don't go outside the app (yet - we might when we reach the true game engine). The idea is to provide a unified success and error response always - so that other API's that invoke ours can have a standardized way of extracting the errors, rather than having to prepare for all possible exceptions thrown